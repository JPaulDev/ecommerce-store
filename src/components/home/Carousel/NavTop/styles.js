import styled from 'styled-components';

export const List = styled.ul`
  background-color: rgb(83 83 83 / 80%);
  border-bottom: 1px solid var(--color-grey-7);
  display: flex;
  justify-content: space-around;
  list-style: none;
  position: absolute;
  top: 0;
  width: 100%;
  z-index: 10;

  @media screen and (max-width: 714px) {
    display: none;
  }
`;

export const Button = styled.button`
  background: none;
  border-bottom: ${({ isActive }) =>
    isActive && '3px solid var(--color-orange-1)'};
  color: #b9b9b9;
  cursor: pointer;
  font-family: 'Uni Sans', sans-serif;
  font-size: 0.83rem;
  font-weight: 600;
  padding: 16px 20px 12px;
  transform: ${({ isActive }) => isActive && 'scale(1.06)'};

  &:hover {
    color: #e5e7eb;
  }

  && {
    color: ${({ isActive }) => isActive && 'white'};
  }
`;
