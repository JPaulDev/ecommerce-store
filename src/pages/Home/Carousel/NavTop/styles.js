import styled from 'styled-components';

export const Container = styled.div`
  background: rgb(83 83 83 / 80%);
  border-bottom: 1px solid var(--color-grey-7);
  position: absolute;
  top: 0;
  width: 100%;
  z-index: 10;
`;

export const List = styled.ul`
  display: flex;
  justify-content: space-around;
  list-style: none;
`;

export const Button = styled.button`
  background: none;
  border-bottom: ${({ isActive }) =>
    isActive && '3px solid var(--color-orange-1)'};
  color: #b9b9b9;
  cursor: pointer;
  font-family: 'Uni Sans', sans-serif;
  font-size: 0.83rem;
  font-weight: 600;
  padding: 16px 20px 12px;
  transform: ${({ isActive }) => isActive && 'scale(1.06)'};

  &:hover {
    color: #e5e7eb;
  }

  && {
    color: ${({ isActive }) => isActive && 'white'};
  }
`;
