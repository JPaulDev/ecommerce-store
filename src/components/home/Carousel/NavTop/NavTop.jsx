import * as Styled from './styles';

export default function NavTop({ slides, slideIndex, onChangeSlide }) {
  return (
    <Styled.List>
      {slides.map((slide, index) => (
        <li key={slide.topNavText}>
          <Styled.Button
            type="button"
            isActive={index === slideIndex}
            onClick={() => onChangeSlide('JUMP', index)}
          >
            {slide.topNavText}
          </Styled.Button>
        </li>
      ))}
    </Styled.List>
  );
}
