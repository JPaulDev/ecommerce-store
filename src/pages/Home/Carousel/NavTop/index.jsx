import * as Styled from './styles';

export default function NavTop({ slides, slideIndex, onChangeSlide }) {
  return (
    <Styled.Container data-testid="nav-top">
      <Styled.List>
        {slides.map((slide, index) => (
          <li key={slide.topNavText}>
            <Styled.Button
              type="button"
              isActive={index === slideIndex}
              onClick={() => onChangeSlide('JUMP', index)}
            >
              {slide.topNavText}
            </Styled.Button>
          </li>
        ))}
      </Styled.List>
    </Styled.Container>
  );
}
