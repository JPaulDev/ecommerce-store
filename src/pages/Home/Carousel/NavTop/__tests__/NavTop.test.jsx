import { screen, render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import NavTop from '../index';

const slides = [
  {
    topNavText: '30 SERIES 3XS PCs',
  },
  {
    topNavText: 'INTEL 3XS PCs',
  },
];

describe('NavTop component', () => {
  it('renders a button with text for each slide in the slides array', () => {
    const tree = renderer.create(<NavTop slides={slides} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('applies styles to the active slide button', () => {
    render(<NavTop slides={slides} slideIndex={0} />);
    const button = screen.getByRole('button', {
      name: /30 series 3xs pcs/i,
    });

    expect(button).toHaveStyle(`
      transform: scale(1.06);
      color: white;
    `);
  });
});
