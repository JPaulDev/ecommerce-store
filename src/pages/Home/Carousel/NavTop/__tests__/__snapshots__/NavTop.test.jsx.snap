// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavTop component renders a button with text for each slide in the slides array 1`] = `
<div
  className="sc-bczRLJ ldEQiT"
  data-testid="nav-top"
>
  <ul
    className="sc-gsnTZi kPFAvu"
  >
    <li>
      <button
        className="sc-dkzDqf kvZhko"
        onClick={[Function]}
        type="button"
      >
        30 SERIES 3XS PCs
      </button>
    </li>
    <li>
      <button
        className="sc-dkzDqf kvZhko"
        onClick={[Function]}
        type="button"
      >
        INTEL 3XS PCs
      </button>
    </li>
  </ul>
</div>
`;
